lib/rubocop/ast/node/def_node.rb

```diff
@@ -13,7 +13,7 @@ class DefNode < Node
       #
       # @return [Boolean] whether the `def` node body is a void context
       def void_context?
-        method?(:initialize) || assignment_method?
+        (def_type? && method?(:initialize)) || assignment_method?
```

Comment:
Conceptually this makes sense to me, though I wonder if would be better to instead introduce a `DefsNode` instead which then overrides this method

Reply:
I actually thought `DefsNode` was already defined going into this.

I'd rather not make drastic changes in this PR. This is a pre-existing pattern (used in `BlockNode` for example):

https://github.com/rubocop/rubocop-ast/blob/30439dbd930ca7a68e83b83fe884226a8747aad8/lib/rubocop/ast/node/block_node.rb#L49

Reply:
Ah, right. Yeah, probably fine then. I did a quick search around but didn't find other nodes that did it like this but clearly I didn't search good enough.